### PR TITLE
[Feature] Mimic add friend & deleting friend

### PR DESCRIPTION
**Pull request by dlgmltjr0925**
## Description

I thought I needed a friend provider to implement add and delete friends. So first, I made a friend provider using the "immer" library presented in the issue, and applied it to the [SearchUser] screen and [Friend] screen.

First, I implemented a profile modal that allows users to add and delete friends when the user is pressed on the [SearchUser] screen. When adding a friend, the message "Added to your friend" will be displayed in the modal, and the user will be added to the Friend Provider's friend list and you will see the friend added in the [Friend] screen. If you delete a friend, the user is deleted from the Friend Provider's friend list, and you can see that the user has been deleted from the [Friend] screen.

Secondly, when a friend is clicked on the [Friend] screen, a profile modal appears to delete the friend. When you delete a friend, you can see that your profile modal disappears and that friend is removed from your friends list.

## Related Issues

#39 

## Tests

I added the following tests:

Friend.test.tsx

- should be added to the friend list when called addFriend function
- should be deleted to the friend list when called deleteFriend function

ProfileModal.test.tsx

- should be shown and closed when press buttons
- should display added friend message when press add button in [SearchUser] Screen
- should close when press delete button in [Friend] Screen

FriendProvider.test.tsx

- [FriendProvider] rendering test
- should be added only once, when add the same user twice
- should be added 2 users, when added 2 unregistered users
- should be deleted, when delete registered friend
- should be deleted 2 user, when delete 2 registered friends

SearchUser.test.tsx

- show profile modal when press user in search user list

## Checklist

Before you create this PR confirms that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide](https://github.com/dooboolab/hackatalk-mobile/blob/master/CONTRIBUTING.md) and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] Run `yarn test` or `yarn test -u` if you need to update snapshot.
- [x] Run `yarn lint`
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

File: src/providers/ProfileModalProvider.tsx [hunk 1]
```diff
@@ -30,22 +36,26 @@ const initialState: State = {
...
   children?: React.ReactElement;
 }
 
 type Reducer = (state: State, action: Action) => State;
 
-const showModal = (dispatch: React.Dispatch<Action>) => (user: Partial<User>, deleteMode: boolean): void => {
+const showModal = (dispatch: React.Dispatch<Action>) => (
+  user: Partial<User>,
+  deleteMode: boolean,
+  screen?: string,
```
Comment (hyochan) [inline src/providers/ProfileModalProvider.tsx]: Try not to give type like this. Use param that could be more commonly used. 

For me, I'd give `onDeleteCallback` type here to customize the callback.

**Comment by hyochan** [comment]
How is this going?

**Comment by dlgmltjr0925** [comment]
@hyochan Sorry. I'm lately. 
But it is in progress. 
As your commented, I have implemented a showModal method that takes a callback as an argument. Only test code modifications remain.
I'll pull request by tomorrow.

File: src/providers/ProfileModalProvider.tsx (hunk 1 above)
Comment (dlgmltjr0925) [reply src/providers/ProfileModalProvider.tsx]: @hyochan I'v made some modifications to showModal's params to write the code as you requested. It takes callback methods called [onAddFriend] and [onDeleteFriend] as parameters of showModal, and handles actions after adding or deleting a friend.

**Comment by hyochan** [comment]
image

@dlgmltjr0925 I don't know why `cla` is pending. Could you kindly reopen the pull request making another branch? You may try to take another branch from current branch.

**Pull request closed without merging**